### PR TITLE
limit class path loading

### PR DESCRIPTION
**Pull request by mgudemann**
This introduces a cleaner version of limiting the class files that are loaded, including both from JAR archives and also from the classpath. The overall goal is to be able to analyze class files from a Java project without any preprocessing.

### CODE REVIEW

**Comment by tautschnig** [comment]
@mgudemann Could you please look into the build failures?

File: src/java_bytecode/java_bytecode_language.cpp [hunk 1]
```diff
@@ -187,7 +190,7 @@ bool java_bytecode_languaget::parse(
     if(main_class.empty())
     {
       status() << "JAR file without entry point: loading it all" << eom;
```
Comment (peterschrammel) [inline src/java_bytecode/java_bytecode_language.cpp]: Is this message still correct?

File: src/java_bytecode/java_class_loader_limit.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+{
+  std::regex regex_matcher;
+  std::set<std::string> set_matcher;
+  bool regex_match;
+  std::smatch string_matcher;
+
+  void setup_class_load_limit(std::string &);
+ public:
+  explicit java_class_loader_limitt(
+    message_handlert _message_handler,
```
Comment (peterschrammel) [inline src/java_bytecode/java_class_loader_limit.h]: this must be a reference

File: src/java_bytecode/java_bytecode_language.cpp (hunk 1 above)
Comment (mgudemann) [inline src/java_bytecode/java_bytecode_language.cpp]: this is `loading class files` now

**Comment by kroening** [comment]
Who would write the class loading limits?

**Comment by mgudemann** [comment]
My idea was to use this to define an analysis context for a project, i.e., loading classes deemed relevant for an anlysis that is precise enough, but does not consume too many resources. In such a case, the limit should be written by a program that calls CBMC which might change iteratively which classes are visible.

**Pull request merged**